src/debugger/android/androidPlatform.ts

```diff
@@ -48,24 +51,20 @@ export class AndroidPlatform implements IAppPlatform {
             () =>
                 Q(AndroidPlatform.RUN_ANDROID_FAILURE_PATTERNS)).process(runAndroidSpawn);
 
-        return output
-            .then(() => pkg.name())
-            .then(appName => new PackageNameResolver(appName).resolvePackageName(runOptions.projectRoot))
-            .then(packageName => {
-                this.packageName = packageName;
-                return this.deviceHelper.getConnectedDevices()
-                    .then((devices: IDevice[]) => {
-                        if (devices.length > 1) {
-                            /* more than one device or emulator */
-                            this.debugTarget = this.getTargetEmulator(runOptions, devices);
-                            if (this.debugTarget) {
-                                /* Launching is needed only if we have more than one device active */
-                                return this.deviceHelper.launchApp(runOptions.projectRoot, packageName, this.debugTarget);
-                            }
-                        } else if (devices.length === 1) {
-                            this.debugTarget = devices[0].id;
-                        }
-                    });
+        return output.finally(() =>
+            this.deviceHelper.getConnectedDevices().then(devices => {
```

Comment:
Even though this is technically a single statement, I would suggest including the braces since it is a bit long and I could see things accidentally breaking if someone adds an additional statement but forgets to add a return